pkg/testutils/mock_config.go

```diff
@@ -31,6 +31,16 @@ func NewFakeClusterConfig(cfgMap *v1.ConfigMap) (*virtconfig.ClusterConfig, cach
 	return virtconfig.NewClusterConfig(configMapInformer, crdInformer, kubeVirtInformer, namespace), configMapInformer, crdInformer, kubeVirtInformer
 }
 
+func NewFakeClusterConfigWithCPUArch(cfgMap *v1.ConfigMap, CPUArch string) (*virtconfig.ClusterConfig, cache.SharedIndexInformer, cache.SharedIndexInformer, cache.SharedIndexInformer) {
+	clusterConfig, configMapInformer, crdInformer, kubeVirtInformer := NewFakeClusterConfig(cfgMap)
+
+	if CPUArch != "" {
+		clusterConfig.SetCPUArch(CPUArch)
```

Comment:
I was hoping to get rid of this `SetCPUArch` all-together. Based on the CPU arch we create different default values. How about this:


Our production code should definitely call:

```go
func NewClusterConfig(configMapInformer cache.SharedIndexInformer,
	crdInformer cache.SharedIndexInformer,
	kubeVirtInformer cache.SharedIndexInformer,
	namespace string) *ClusterConfig
```

But what about adding a

```go
func NewClusterConfigWithCPUArch(configMapInformer cache.SharedIndexInformer,
	crdInformer cache.SharedIndexInformer,
	kubeVirtInformer cache.SharedIndexInformer,
	namespace string, cpuArch string) *ClusterConfig
```

function which accepts the architecture as an argument?

`NewClusterConfig` could then look like this:

```go
func NewClusterConfig(configMapInformer cache.SharedIndexInformer,
	crdInformer cache.SharedIndexInformer,
	kubeVirtInformer cache.SharedIndexInformer,
	namespace string) *ClusterConfig {
	return NewClusterConfigWithCPUArch(
	    configMapInformer, 
	    crdInformer, 
	    kubeVirtInformer, 
	    namespace, 
	    runtime.GOARCH,
	    )
	
	}
```

We can then do the same for the fake cluster config functions.

This has the main advantage that we always get a clean fully defaulted function and that we can get rid of the `SetCPUArch` method.